FitnessClub/Data/BLL/Services/CoachService.cs

```diff
@@ -0,0 +1,135 @@
+﻿using FitnessClub.Data.BLL.Interfaces;
+using FitnessClub.Data.DAL.Interfaces;
+using FitnessClub.Data.DAL.Utility;
+using FitnessClub.Data.Models.ViewModels;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace FitnessClub.Data.BLL.Services
+{
+    public class CoachService : EmployeeService, ICoachService
+    {
+        private readonly ISessionRepository sessionRepository;
```

Comment:
Is there a reason why is sessionRepository not protected in EmployeeService so that it can be visible here, in CoachService rather than injecting a separate instance?

Reply:
I also thought that's not the best idea, and redone this class a while ago. Currently, CoachService does not inherit from EmployeeService, so the problem is already solved. The other reason to do that was, with inheritence the base constructor call was really ugly. In new implementation (which already is commited on the dev branch), if a page, presumably a control panel for a coach, requires to access both Employee and CoachService, it would just inject both services separately.